cli/command/environment_test.go

```diff
@@ -338,19 +296,93 @@ func TestEnvironmentSetMinCount(t *testing.T) {
 	})
 }
 
-func TestEnvironmentLink(t *testing.T) {
+func TestEnvironmentLinkBiDirectional(t *testing.T) {
 	testWaitHelper(t, func(t *testing.T, wait bool) {
 		base, ctrl := newTestCommand(t)
 		defer ctrl.Finish()
 
 		command := NewEnvironmentCommand(base.Command())
 
-		job := &models.Job{
-			JobID:  "job_id",
+		links1 := []string{"env_id2"}
+		links2 := []string{"env_id3", "env_id1"}
+
+		base.Resolver.EXPECT().
+			Resolve("environment", "env_name1").
+			Return([]string{"env_id1"}, nil)
+
+		base.Resolver.EXPECT().
+			Resolve("environment", "env_name2").
+			Return([]string{"env_id2"}, nil)
+
+		base.Client.EXPECT().
+			ReadEnvironment("env_id1").
+			Return(&models.Environment{
+				EnvironmentID: "env_id1",
+				Links:         []string{},
+			}, nil)
+
+		base.Client.EXPECT().
+			ReadEnvironment("env_id2").
+			Return(&models.Environment{
+				EnvironmentID: "env_id2",
+				Links:         []string{"env_id3"},
+			}, nil)
+
+		job1 := &models.Job{
+			JobID:  "job_id1",
 			Status: job.Completed.String(),
-			Result: "entity_id",
+			Result: "env_id1",
+		}
+
+		req1 := models.UpdateEnvironmentRequest{EnvironmentID: "env_id1"}
+		req1.Links = &links1
+
+		base.Client.EXPECT().
+			UpdateEnvironment(req1).
```

Comment:
Updates should take `entityID string` as first param.

Reply:
That's new. That would require changing the client interface for all the entities.